source/uri-options/uri-options.rst

```diff
@@ -88,8 +88,9 @@ pertaining to URI options apply here.
    * - authMechanism
      - any string; valid values are defined in the `auth spec 
        <https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#supported-authentication-methods>`_
-     - defined in `auth spec 
-       <https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#supported-authentication-methods>`_
+     - None; default values for authentication exist for constructing authentication credentials per the
+       `auth spec <https://github.com/mongodb/specifications/blob/master/source/auth/auth.rst#supported-authentication-methods>`_,
+       but must not be assigned here to distinguish between values specified by the user and defaulted by the implementation.
```

Comment:
I think the tests are a bit vague about what the 'auth' field is supposed to mean.  If it's just "stuff I found in the string".  There's no test for `authSource` in valid-auth.yml and no where for it to go in the 'auth' field.

I'm hesitant to revise the test format and make work for everyone.  If the current format works and we're just clarifying, then maybe that's enough.  What do you think?

Reply:
The problem I have with this clause of the sentence is that it introduce a distinction between two  undefined terms: "here" and "the implementation" with no guidance about what that actually means.  Then it uses a lower-case "must".  And there are no tests to clarify the distinction.

I suggest just striking everything after the last comma.

Reply:
The more I've considered this, I don't think the *auth* tests are sufficient in their current structure, so I'm experimenting with making them more specific to translating from a URI to an auth credential to provide a stronger set of assertions.  I think the connection string and URI options tests can remain as is as tests of pure parsing output.  I'll post a new commit when I finish my experiment.

Reply:
I reworded the text in the latest commit to avoid vague "must not".  Now it's just a statement to reiterate there is no default.  The URI options tests aren't expressive enough to test it, but it's effectively tested by the new auth tests that translate from URI to credential object.